### ISSUE TITLE
Upgrade to Kafka 0.10.2.0

### ISSUE THREAD

**Issue by mikisseth** [jcustenborder/kafka-connect-spooldir#18]
The current latest release on github is based on Kafka 0.10.0.0. It would be nice to have releases for Kafka 0.10.1.0 and Kafka 0.10.2.0.

**Comment by mikisseth** [comment]
A lot of work have been done on the schema refactoring branch, so it would be natural to base the release on the code here. Below you will find a rather length list of problems I've encountered with the the current schema-refactor HEAD build:

1.  Build fails with a 
`[ERROR] Failed to execute goal on project kafka-connect-spooldir: Could not resolve dependencies for project com.github.jcustenborder.kafka.connect:kafka-connect-spooldir:jar:1.0-SNAPSHOT: Could not find artifact org.yaml:snakeyaml:jar:1.19-SNAPSHOT`
this is caused by the pom defining a dependency to jfairy, which has a dependency to snakeyaml declared with a range version `<version>[1.9,2.0)</version>`. Together with MNG-3092 this causes  Maven to try to download a non-existing snapshot. 
My workaround was to add a specific version to a dependency management section as 
```
   <dependencyManagement>
        <dependencies>
            <!-- Avoid downloading non-existing snapshot dependencies cause by jfairy range version definition for snakeyaml dependency -->
            <dependency>
                <groupId>org.yaml</groupId>
                <artifactId>snakeyaml</artifactId>
                <version>1.18</version>
            </dependency>
        </dependencies>
    </dependencyManagement>
```
2. A number of tests are failing.  Maven reports: 
* 5 x SpoolDirCsvSourceTaskTest.lambda$null$0:54->SpoolDirSourceTaskTest.poll:138
* 4 x SpoolDirJsonSourceTaskTest.lambda$null$0:54->SpoolDirSourceTaskTest.poll:138
on my computer
3. I can not compare my build status to a reference build, as no CI is available. A Jenkins file is present, which implies that CI exists somewhere. Would it be possible to switch to travis, so the CI status was general available?
4. I miss the automatic Avro schema generation from the master branch :-(

**Comment by mikisseth** [comment]
Note that if I might be able to help with some of the stuff mentioned above, if I could get a pointer on how to proceed.

**Comment by jcustenborder** [comment]
@mikisseth Thanks for the feedback!

> 1. Build fails with a

I'll take a look at this. Thanks!

> 2. A number of tests are failing. Maven reports:

I'll take a look at this. Thanks!

>3.  I can not compare my build status to a reference build, as no CI is available. A Jenkins file is present, which implies that CI exists somewhere. Would it be possible to switch to travis, so the CI status was general available?

I can look at doing something like this in the future. For now I have my own build process that runs on jenkins in my home lab. It's what builds the RPM, DEB, and docker images. 

> I miss the automatic Avro schema generation from the master branch

How were you using this? I was thinking about introducing this functionality again through an additional tool. For example you would point the tool at a file and it'll output the suggested schema. I see the usefulness of the original feature, but I was hoping that setting the schema as a job setting would help people do more type conversion at the connector level.

**Comment by mikisseth** [comment]
@jcustenborder 

Sounds good :-)

Regarding the Avro schema generation functionality, I was hoping to make it as easy possible for a (non-technical) user of a Kafka + Connect server + [Connect UI](https://github.com/Landoop/kafka-connect-ui) to add a new ` kafka-connect-spooldir` source connector through the UI without having to understand how to defining and registering a Avro schema (and still get a simple schema registered). This should of course be an optional setting, so more advanced users may define explicit and more precise schemas (which might be based on a tool generated 'draft schema').

**Comment by jcustenborder** [comment]
@mikisseth Can you take another look?

I gotta say I really don't like the auto schema generation feature. My point of this connector was hoping to get people to import strongly typed data into kafka. Man I want an integer to be an integer. ;) But whatever. 

I added the feature back. It's enabled with the `schema.generation.enabled` configuration setting. As my sign of protest it defaults to `false` haha. It forces all files for a particular pattern to have the same schema. For example `^test\.*\.csv$` must have the same fields.

**Comment by mikisseth** [comment]
Some initial feedback.

With the JFairy dependency gone I compile without problems.
The `schema.generation.enabled` option was exactly what I was hoping for my rather simplistic initial step to configure a connector supported by a (draft) schema :-).

The tests are still failing with: 
`value() does not match.: last_login does not match. ==> expected: <Thu Oct 01 00:29:03 GST 2015> but was: <Wed Sep 30 15:29:03 GST 2015>`
my guess is that this is because the tests are time zone dependent.

I tried running the tests in IntelliJ, but failed because the JUnit5 version was `5.0.0-M3`. This had me a bit confused as this was inherited from the `com.github.jcustenborder.kafka.connect:kafka-connect-parent:0.10.2.0-cp1` pom which declares the JUnit5 version as `5.0.0-M4`. I found out that even though the `kafka-connect-parent 0.10.2.0-cp1` is a non-snapshot version, the pom for this specific version changes. This means Mavens normal assumption for non-snapshot versions (they are considered releases, and therefor never change) breaks down, and I needed to manually remove `kafka-connect-parent` from the .m2 local repository to get the parent pom updated and thereby get the correct `5.0.0-M4`dependency.

I'll try to spin up the connector in our Kafka setup next week and get back to you.
